### PR TITLE
cephadm: add ability to continue on failure when applying multiple specs

### PR DESCRIPTION
**Pull request by adk3798**
Additionally, add the flag that does so when cephadm applies a spec during bootstrap. Bootstrap will continue to completion regardless of whether applying the spec fails, so we might as well try applying all of it while reporting errors we do see back to the user

Fixes: https://tracker.ceph.com/issues/65338

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-09-10_02:59:23-orch:cephadm-wip-adk-testing-2024-09-09-1904-distro-default-smithi/

Failures:
- stray daemon warning in smb test, known issue
- `Test failure: test_nfs_export_creation_without_cmount_path (tasks.cephfs.test_nfs.TestNFS)` caused by a PR in the run
- 2x agent on staggered upgrade failure, known issue
- deploy raw test timed out, known issue
- `test_rgw_multisite` test failed to generate token on realm bootstrap, somewhat new but not the first time I've seen it and it's unrelated to PRs in this run
- 4x mds_upgrade_sequence failures, known issue
- rgw-ingress tests timed out, known issue

Nothing to block merging PRs in the run other than the NFS failure related one

**Comment by adk3798** [comment]
jenkins test make check

**Review by phlogistonjohn — APPROVED** [review]
A smidge hard to follow but it seems to do the trick as far as I can tell.

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-09-18_05:05:01-orch:cephadm-wip-adk-testing-2024-09-17-1139-distro-default-smithi/

Failures:
- nfs task failed due to PR in the run
- 3x mds_upgrade sequence failures, known issue
- 2x SMB test Selinux denials, known issue
- staggered upgrade with agent enabled failure, known issue
- 2x infra/teuth failures `Permission denied (publickey)`, `Failed to reconnect to smithi179`
- rgw-ingress tests timing out, known issue

Only PR in the run that can't be merged based off the results is the one causing the nfs test failure

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
https://pulpito.ceph.com/phlogistonjohn-2024-10-15_13:29:56-orch:cephadm-wip-adk-testing-2024-10-14-1530-distro-default-smithi/

Failures:
- 3x failures in smb ctdb tests due to either stray daemon warnings or selinux denials, known issue
- 3x misc. infra/teuthology failures
- 3x mds_upgrade sequence timeouts or cluster log warnings, known issue
- 2x rgw ingress test timeouts, known issue
- mgr-nfs-upgrade failure, looks like it tried to check all the daemons were upgraded while only the first mgr had been upgraded. It looks like maybe something went wrong with the check to make sure the upgrade had completed.

Would maybe hold off on merging upgrade related PRs, but everything else should be okay.
